src/from-promise.ts

```diff
@@ -207,10 +207,10 @@ export namespace fromPromise {
         return p
     })
 
+    function resolveBase<T>(value: T): IFulfilledPromise<T> & IBasePromiseBasedObservable<T>
     function resolveBase<T>(
         value?: T
     ): IFulfilledPromise<T | undefined> & IBasePromiseBasedObservable<T>
-    function resolveBase<T>(value: T): IFulfilledPromise<T> & IBasePromiseBasedObservable<T>
```

Comment:
Here is the bug fix: moving the overload up one line, so it is given priority by the type-checker.